pkg/scheduler/plugins/prioritize/priorities/image_locality.go

```diff
@@ -0,0 +1,104 @@
+/*
```

Comment:
why do we need this file?

Reply:
ImageLocalityPriority Function expects priorityMetadata struct as argument, to create that struct we need PriorityMetadataFactory struct, for which we need ServiceLister, ControllerLister, ReplicaSetLister, StatefulSetLister interface to be implemented, but ImageLocalityPriority Function only need totalNumNodes only, so have modified function that it expects only totalNumNodes alone.

Reply:
Do not copy it or modify it; we do not plan effort to keep the compatbility with upstream.

Reply:
Then we will implement our own logic to calculate score for image locality priority, because we will not be able to create PriorityMetadataFactory, without that priorityMetadata struct cannot be also created, so we cannot use image locality priority function provided by kubernetes.

Reply:
How do you want me to handle it?

Reply:
Open an issue in upstream to publish `priorityMetadata`; remove it from this PR right now, and leave a `TODO` there.